Product/Production/Services/DocumentQueryCore/src/main/java/gov/hhs/fha/nhinc/docquery/deferred/impl/AdapterDeferredResponseOptionImpl.java

```diff
@@ -0,0 +1,54 @@
+/**
+ *
+ */
+package gov.hhs.fha.nhinc.docquery.deferred.impl;
+
+import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.common.nhinccommonadapter.AdapterDeferredResponseOptionQuerySecuredType;
+import gov.hhs.fha.nhinc.common.nhinccommonadapter.AdapterDeferredResponseOptionQueryType;
+import gov.hhs.fha.nhinc.docrepository.adapter.dao.DeferredXCARequestDao;
+import gov.hhs.fha.nhinc.docrepository.adapter.model.DeferredXCARequest;
+import gov.hhs.fha.nhinc.wsa.WSAHeaderHelper;
+import oasis.names.tc.ebxml_regrep.xsd.query._3.AdhocQueryRequest;
+
+/**
+ * @author ptambellini
+ *
+ */
+public class AdapterDeferredResponseOptionImpl {
+
+    public AdapterDeferredResponseOptionQueryType respondingGatewayCrossGatewayQuery(AdhocQueryRequest msg,
+        AssertionType assertion) {
+
+        String newId = saveRequest(msg, assertion);
+        AdapterDeferredResponseOptionQueryType response = new AdapterDeferredResponseOptionQueryType();
+        response.setNewId(newId);
+        response.setAssertion(assertion);
+        return response;
+    }
+
+
+
+    public AdapterDeferredResponseOptionQuerySecuredType respondingGatewayCrossGatewayQuerySecured(
+        AdhocQueryRequest msg,
+        AssertionType assertion) {
+
+        String newId = saveRequest(msg, assertion);
+        AdapterDeferredResponseOptionQuerySecuredType response = new AdapterDeferredResponseOptionQuerySecuredType();
+        response.setNewId(newId);
+        return response;
+    }
+
+    private static String saveRequest(AdhocQueryRequest msg, AssertionType assertion) {
```

Comment:
Optional: this method is just confusing. We are calling it SaveRequest but a new Id is generated and returned here.